lib/modules/datasource/npm/get.ts

```diff
@@ -140,7 +178,13 @@ export async function getDependency(
       (whitelistedPublicScopes.includes(packageName.split('/')[0]) ||
         !packageName.startsWith('@'))
     ) {
-      await packageCache.set(cacheNamespace, packageUrl, dep, cacheMinutes);
+      const cacheData = { softExpireAt, etag };
+      await packageCache.set(
+        cacheNamespace,
+        packageUrl,
+        { ...dep, cacheData },
+        hardExpireMinutes
```

Comment:
we should use softExpire if no etag
```suggestion
        etag ? hardExpireMinutes : cacheMinutes
```